tools/crm_resource_runtime.c

```diff
@@ -188,18 +189,21 @@ find_matching_attr_resource(resource_t * rsc, const char * rsc_id, const char *
                 break;
         }
 
-    } else if (rsc->parent && BE_QUIET == FALSE) {
+    } else if(rsc->parent && BE_QUIET == FALSE) {
         printf("Forcing %s of '%s' for '%s' instead of '%s'\n", cmd, attr_name, rsc_id, rsc->parent->id);
 
-    } else if(rsc->parent == NULL && rsc->children) {
-        resource_t *child = rsc->children->data;
+    } else if(rsc->parent == NULL) {
+        for(GList *gIter = rsc->children; gIter; gIter = gIter->next) {
+            resource_t *child = gIter->data;
 
-        if(child->variant == pe_native) {
+            if(child->variant != pe_native) {
+                continue;
+            }
             lookup_id = clone_strip(child->id); /* Could be a cloned group! */
             rc = find_resource_attr(cib, XML_ATTR_ID, lookup_id, attr_set_type, attr_set, attr_id, attr_name, &local_attr_id);
 
             if(rc == pcmk_ok) {
-                rsc = child;
+                resources = g_list_append(resources, child);
```

Comment:
If I'm following this correctly, we'll only set the attribute on the child if it already has an explicit value. That makes sense.

Reply:
Yes, this is correct.